public/icvar.h

```diff
@@ -11,21 +11,254 @@
+	int64_t		m_i64Value;
+	uint64_t	m_u64Value;
+	float		m_flValue;
+	double		m_dbValue;
+	const char*	m_szValue;
+	Color		m_clrValue;
+	Vector2D	m_vec2Value;
+	Vector		m_vec3Value;
+	Vector4D	m_vec4Value;
+	QAngle		m_angValue;
+};
+static_assert(sizeof(float) == sizeof(int32_t), "Wrong float type size for ConVar!");
+static_assert(sizeof(double) == sizeof(int64_t), "Wrong double type size for ConVar!");
+
+template<> inline CVValue_t& CVValue_t::operator=<bool>( const bool other )					{ m_bValue = other; return *this; }
+template<> inline CVValue_t& CVValue_t::operator=<int16_t>( const int16_t other )			{ m_i16Value = other; return *this; }
+template<> inline CVValue_t& CVValue_t::operator=<uint16_t>( const uint16_t other )			{ m_u16Value = other; return *this; }
+template<> inline CVValue_t& CVValue_t::operator=<int32_t>( const int32_t other )			{ m_i32Value = other; return *this; }
+template<> inline CVValue_t& CVValue_t::operator=<uint32_t>( const uint32_t other )			{ m_u32Value = other; return *this; }
+template<> inline CVValue_t& CVValue_t::operator=<int64_t>( const int64_t other )			{ m_i64Value = other; return *this; }
+template<> inline CVValue_t& CVValue_t::operator=<uint64_t>( const uint64_t other )			{ m_u64Value = other; return *this; }
+template<> inline CVValue_t& CVValue_t::operator=<float>( const float other )				{ m_flValue = other; return *this; }
+template<> inline CVValue_t& CVValue_t::operator=<double>( const double other )				{ m_dbValue = other; return *this; }
+template<> inline CVValue_t& CVValue_t::operator=<const char*>( const char* other )			{ m_szValue = other; return *this; }
+template<> inline CVValue_t& CVValue_t::operator=<const Color&>( const Color& other )		{ m_clrValue = other; return *this; }
+template<> inline CVValue_t& CVValue_t::operator=<const Vector2D&>( const Vector2D& other )	{ m_vec2Value = other; return *this; }
+template<> inline CVValue_t& CVValue_t::operator=<const Vector&>( const Vector& other )		{ m_vec3Value = other; return *this; }
+template<> inline CVValue_t& CVValue_t::operator=<const Vector4D&>( const Vector4D& other )	{ m_vec4Value = other; return *this; }
+template<> inline CVValue_t& CVValue_t::operator=<const QAngle&>( const QAngle& other )		{ m_angValue = other; return *this; }
+
+struct CSplitScreenSlot
+{
+	CSplitScreenSlot( int index )
+	{
+		m_Data = index;
+	}
+	
+	int Get() const
+	{
+		return m_Data;
+	}
+	
+	int m_Data;
+};
+
+//-----------------------------------------------------------------------------
+// Called when a ConVar changes value
+//-----------------------------------------------------------------------------
+typedef void(*FnChangeCallbackGlobal_t)(BaseConVar* ref, CSplitScreenSlot nSlot, const char *pNewValue, const char *pOldValue);
+
+//-----------------------------------------------------------------------------
+// Purpose: Replaces the ConVar*
+//-----------------------------------------------------------------------------
+class ConVarHandle

```

Comment:
As per discussion in discord, split this back into 2 handles for convar and concommand for better distinction between both on usage.